### ISSUE TITLE
Why network that have worked in version 0.1.1 does not work in 1.0.0?

### ISSUE THREAD

**Issue by pi-null-mezon** [tiny-dnn/tiny-dnn#332]
When I migrate to the version 1.0.0 I have found strange behavior. My network that i previously trained by version 0.1.1 of tiny-dnn (I have implement SegNet-like architecture for Chest X-RAY lungs segmentation and it works surprisingly well) do not produce similar results with tiny-dnn version 1.0.0. In v.0.1.1 my network trains well (I can see that at the each new epoch of training the loss function value becomes lower and lower and the activation in the last layer becomes more and more accurate), but in v.1.0.0 training process on the same data seems to diverges all the time (loss is not changed considerably between epoches and the last layer activation is not converges to lungs-like projection) . So why it is could happen? Also, I should tell, that I have not use built-in serialization method and instead load weights by my own (and when I rewrite code for the v.1.0.0 support one thing I have done was replacement of get_weights() method to weights() and call init_weights() after network creation and weights upload). Sorry for such long description

**Comment by nyanp** [comment]
@pi-null-mezon 
Since tiny-dnn have changed it's parallelism strategy, the result could be differ. What happens if you change the learning rate a bit smaller (x0.8 or x0.5) in your optimizer?

**Comment by pi-null-mezon** [comment]
I have tried x0.8, x0.5, x0.1 with no success.

**Comment by bhack** [comment]
@nyanp Code is at https://github.com/pi-null-mezon/OpenIST/blob/master/CMDUtils/SegNet/cnnsegmentnet.cpp but I don't think that the dataset is available to reproduce exactly the same experiment.

**Comment by pi-null-mezon** [comment]
I will just add some images, that explain problem.
So this is the raw image: 
![image](https://cloud.githubusercontent.com/assets/5991537/19559381/4215330a-96d8-11e6-90ee-a4b84c77edb5.png);
I want to segment lungs. Using SegNet architecture and tiny-cnn v0.1.1 I have get following result: 
![image](https://cloud.githubusercontent.com/assets/5991537/19559431/7af39cc0-96d8-11e6-8140-0e2a622ef29c.png);
All seems to be fine. But after iny-dnn update to the version 1.0.0 same network starts to produce bad results:
![image](https://cloud.githubusercontent.com/assets/5991537/19559489/cc807c16-96d8-11e6-9bfe-b41338081c78.png)
Ok, I have said, let's retrain network and perform an experiment. If in v.0.1.1 convergence has been seen after few epoches in v.1.0.0 training procedure does not converge and even after 30-50 epoches produce random pictures like this: 
![image](https://cloud.githubusercontent.com/assets/5991537/19559607/4a8862f4-96d9-11e6-903f-df74dcec665b.png)

**Comment by pi-null-mezon** [comment]
Dataset is available here: https://github.com/pi-null-mezon/OpenIST/tree/master/Datasets

**Comment by bhack** [comment]
Thanky now we have all the info to reproduce your experiment and check if there is any regression.

**Comment by pi-null-mezon** [comment]
I should note feaw little things. I have trained network with input size 128x128 pixels with 4 samples per minibatch - it is not very interesting, just to clarify. But what is very important that in success experiments float_t type was defined as double. When I made experiments with float_t as float  network "blows up" (nan in weights) after 5-7 epoches.

**Comment by nyanp** [comment]
@pi-null-mezon 
Thank you very much for providing more info, I'll check the difference later. I've also opened a PR #353 to providing double-precision support in v1.0.0. Hope this would help you.

**Comment by pi-null-mezon** [comment]
Hello! Is there any news?

**Comment by nyanp** [comment]
@pi-null-mezon
hi, I couldn't try to reproduce your problems, but now we have a new v1.0.0a2 release. This includes many bug fixes with double precision support. You can retry this :)

**Comment by pi-null-mezon** [comment]
Thanks, I will try.

**Comment by pi-null-mezon** [comment]
Unfortunately new version did not solve the issue. So, I'll ask another question: whether this particular task is the only example of bad performance on new library version?

**Comment by nyanp** [comment]
@pi-null-mezon 
I've confirmed that your code doesn't work on v1.0.0a2. I've noticed that this is the bug around convolutional layer with ```padding::same``` mode. I'll fix it.

**Comment by nyanp** [comment]
@pi-null-mezon 
I've confirmed that 391d26f solved the issue with half learning rate and double precision :)

**Comment by pi-null-mezon** [comment]
I,ve checked branch "fix/padding_same" and my segment network realy works as expected! Thanks for the support!
